### ISSUE TITLE
Adding a new field with a default does not set the default

### ISSUE THREAD

**Issue by svivian** [bolt/bolt#7265]
If you add a new field to an existing ContentType, Bolt does not set that field to the default. For example add this field:

```
level:
    type: integer
    default: 1
```

Then upgrade the database. Existing ContentTypes should have 1 as the level because that's the default, but they don't. They are all set to 0, and the table has 0 as the default in MySQL.

Details
-------

 - Relevant Bolt Version: 3.4.3
 - Install type: Composer
 - PHP version: 7.1
 - Used web server: Apache

**Comment by GawainLynch** [comment]
This is actually not supported by DBAL, as it isn't portable across platforms let alone the utter disaster that is MySQL and it's half-sibling MariaDB.

**Comment by bobdenotter** [comment]
@svivian That's not how the default option works. It's set when you _create_ a new record of a contenttype, as described in the docs ;-)

> When you want to give a record a default value, use `default:`. For most fields this will set the initial value of the field, when _you're creating a new record_ of this ContentType. 

Closing this, since it's not a bug. Feel free to re-open, if you feel it needs more discussion.

**Comment by svivian** [comment]
> it isn't portable across platforms

Really? Default values are in every DB I've used (MySQL, MSSQL, Postgres). Always assumed it was part of the standard.

> It's set when you create a new record of a contenttype

Sure, so by that logic if you add fields to existing types, the existing entries are now as if they were entered with the default value for that field. So it would make sense to set a default (via code if it's not possible with SQL).

BTW there is some inconsistency as integers are set to `NOT NULL` unlike text fields. I think I reported a bug before where text fields could not be added due to no default being set. Don't know if that might cause issues with some other DBMS.
